ddtrace/sampler.py

```diff
@@ -313,11 +319,15 @@ def sample(self, span):
                 break
         else:
             # No rules matches so use agent based sampling
-            if self._agent_sampler.sample(span):
-                self._set_priority(span, AUTO_KEEP)
+            _sampler = self._agent_sampler._get_sampler(span)
+            # TODO: We need to determine the proper SamplingMechanism here
+            # TODO: Do we keep this extracted logic or should we move this logic into RateByServiceSampler.sample ?
+            span.set_metric(SAMPLING_AGENT_DECISION, _sampler.sample_rate)
+            if _sampler.sample(span):
+                self._set_priority(span, AUTO_KEEP, SamplingMechanism.AGENT, _sampler.sample_rate)
```

Comment:
I don't love this.

I think the proper call here is to move what `self._set_priority` is trying to do into a base class for all samplers and then just delegate 100% of this logic to `RateByServiceSampler` / `RateSampler`

It seems like either that or duplicate the `RateByServiceSampler` logic here, which doesn't feel right.